File: 0x06-python-classes/5-square.py
#!/usr/bin/python3
"""Class square create one object with an size define

Raises:
    TypeError: [When size is different of int]
    ValueError: [When size is less that Zero]
    TypeError: [When value is different of int]
    ValueError: [When value is less that Zero]

Returns:
    [Int] -- [attribute private size to public and total area of an square]
"""


class Square:
    """class constructor"""
    def __init__(self, size=0):
        if type(size) is not int:
            raise TypeError('size must be an integer')
        elif size < 0:
            raise ValueError('size must be >= 0')
        self.__size = size

    """class getter"""
    @property
    def size(self):
        return self.__size

    """class setter"""
    @size.setter
    def size(self, value):
        if type(value) is not int:
            raise TypeError('size must be an integer')
        elif value < 0:
            raise ValueError('size must be >= 0')
        self.__size = value

    """class for obtain total area of the square"""
    def area(self, __size=0):
        return self.__size * self.__size

    """method for print object based in his size"""
    def my_print(self):
        if self.__size == 0:
            print()
        else:
            for sqr in range(self.__size):
                for icon in range(self.__size):
                    print('#', end="")
                print()

File: 0x06-javascript-classes/5-square.js
#!/usr/bin/node

const { thistle } = require("color-name");

module.exports = class Square {
  constructor (size = 0) {
    try {
      if (!Number.isInteger(size)) throw 'size must be an integer';
      else if (size < 0) throw 'size must be >= 0';
      this._size = size;
    } catch (err) {
      console.log(err);
    }
  }

  area () {
    return this._size * this._size;
  }

  get size () {
    return this._size;
  }

  set size (value) {
    try {
      if (!Number.isInteger(value)) throw 'size must be an integer';
      else if (value < 0) throw 'size must be >= 0';
      this._size = value;
    } catch (err) {
      console.log(err);
    }
  }

  print () {
    if (this._size === 0) { console.log(" "); }
    else {
      for (let i = 0; i < this._size; i++) {
        console.log('#'.repeat(this._size));
      }
    }
  }
};
